lib/srv/desktop/audit.go

```diff
@@ -29,487 +30,465 @@ import (
 	"github.com/gravitational/teleport/lib/tlsca"
 )
 
-func (s *WindowsService) onSessionStart(ctx context.Context, recorder libevents.SessionPreparerRecorder, id *tlsca.Identity, startTime time.Time, windowsUser, sessionID string, desktop types.WindowsDesktop, err error) {
-	userMetadata := id.GetUserMetadata()
-	userMetadata.Login = windowsUser
+// desktopSessionAuditor is used to build session-related events
+// which are emitted to Teleport's audit log
+type desktopSessionAuditor struct {
```

Comment:
This is the new type which holds on to a lot of common state (so that we don't have to pass these values around every time we want to emit an event).

The other change worth noting is this thing only build events - it's up to the caller to decide whether the event gets emitted to the audit log, recorded to the session recording, or both.